aggregator/src/aggregator.rs

```diff
@@ -3687,6 +3700,23 @@ where
         .boxed()
 }
 
+fn compose_common_wrappers_hack<F, T>(
+    filter: F,
+    cors: Cors,
+    response_time_histogram: Histogram<f64>,
+    name: &'static str,
+) -> BoxedFilter<(impl Reply,)>
+where
+    F: Filter<Extract = (Result<T, Error>,), Error = Rejection> + Clone + Send + Sync + 'static,
+    T: Reply + 'static,
+{
+    filter
```

Comment:
I'm not sure how to proceed here. The existing `compose_common_wrappers` is very carefully designed to only apply the response filter (and especially the response time histogram) _after_ the routing filter has been applied. But our routing filter now extracts some values, so we can't do that quite as easily. I'm getting nowhere trying to convince warp to make this work, and I'm wondering if it's worth the effort if we want to move to Trillium anyway.

Reply:
Hard approach: We could partly get around this if we rewrite our higher order wrappers in terms of `warp::generic::Func` instead of `Filter`, so we can glue things together with `.then()` instead of `.and()`, which means we could line up the routing filter's `Extract` type with the response `Func`'s argument. However, that would then prevent us from using filters in the response half, which is a huge impediment. Maybe the only thing that would work is a higher order combiner that takes a routing filter, a responding filter for extracting additional data, and a responding `Func` that takes a big tuple as an argument, combining both the routing filter's and the responding filter's `Extract` types. We're once again running up against the limitations of the sealed `Filter` trait.

The `http` crate has a feature called "extensions", which lets you store extra data hanging off of a `Request` and/or `Response`, indexed by its type. Warp does have a filter to fetch items out of `Extensions`, but it appears there's no filter to set them, and no way to directly access the `Request` to store them, so that's a dead end.

Easier approach: We could resolve the double-counting in metrics and tracing spans by duplicating our routing filters. The first copy would just match URL patterns, but not extract anything, and be used on the outside of the wrapping filters. (we might have to use `warp::filters::path::peek` for this) The second copy would be integrated with the rest of the responding filter, and work as usual.

Reply:
I think `warp::path::peek` can work, at the cost of some awkward path matching code we have to write. Thanks for the suggestions!